agent/agent.go

```diff
@@ -328,16 +328,45 @@ func (a *agent) reportMetadataLoop(ctx context.Context) {
 	// baseInterval to run.
 	var flight trySingleflight
 
+	postMetadata := func(mr metadataResultAndKey) {
+		lastCollectedAts[mr.key] = mr.result.CollectedAt
+		err := a.client.PostMetadata(ctx, mr.key, *mr.result)
+		if err != nil {
+			a.logger.Error(ctx, "agent failed to report metadata", slog.Error(err))
+		}
+	}
+	flushAllMetadata := func() {
+		wg := sync.WaitGroup{}
+		defer wg.Wait()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case mr := <-metadataResults:
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					postMetadata(mr)
+				}()
+				continue
+			default:
+				return
+			}
+		}
+	}
+
 	for {
+		// Ensure all backpressured metadata is posted.
+		if len(metadataResults) > 1 {
+			flushAllMetadata()
```

Comment:
This could perhaps be moved to after the `select` below (i.e. when `baseTicker.C` is triggered). For now it seems to be doing the exact same thing as the select, except when `baseTicker.C` is triggered. (We probably don't need the `len` check either.)

Reply:
it doesn't really seem to be necessary, so i've just removed it for now